plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/LoginUtils.kt

```diff
@@ -0,0 +1,207 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.openapi.progress.ProcessCanceledException
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFileManager
+import org.slf4j.LoggerFactory
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider
+import software.amazon.awssdk.profiles.Profile
+import software.amazon.awssdk.profiles.ProfileProperty
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.ssooidc.model.InvalidGrantException
+import software.amazon.awssdk.services.ssooidc.model.InvalidRequestException
+import software.amazon.awssdk.services.ssooidc.model.SsoOidcException
+import software.amazon.awssdk.services.sts.StsClient
+import software.aws.toolkits.core.region.AwsRegion
+import software.aws.toolkits.core.utils.error
+import software.aws.toolkits.core.utils.tryOrNull
+import software.aws.toolkits.jetbrains.core.AwsClientManager
+import software.aws.toolkits.jetbrains.core.credentials.profiles.SsoSessionConstants
+import software.aws.toolkits.jetbrains.core.credentials.sono.SONO_REGION
+import software.aws.toolkits.jetbrains.core.credentials.sono.SONO_URL
+import software.aws.toolkits.jetbrains.core.credentials.sso.bearer.InteractiveBearerTokenProvider
+import software.aws.toolkits.jetbrains.utils.runUnderProgressIfNeeded
+import software.aws.toolkits.resources.message
+import software.aws.toolkits.telemetry.CredentialSourceId
+import java.io.IOException
+
+private val LOG = LoggerFactory.getLogger("LoginUtils")
+
+sealed class Login {
```

Comment:
interface?

Reply:
btw Login class is basically extraction of `SetupAuthDialog.doOkAction()`